components/layout/query.rs

```diff
@@ -228,6 +214,21 @@ pub fn process_resolved_style_request(
             _ => return computed_style(Some(fragment)),
         };
 
+        // For line height, the resolved value is the computed value if it
+        // is "normal" and the used value otherwise.
+        if longhand_id == LonghandId::LineHeight {
+            let font = style.get_font();
+            let font_size = font.font_size.computed_size();
+            return match font.line_height {
+                // There could be a fragment, but it's only interesting for `min-width` and `min-height`,
+                // so just pass None.
+                LineHeight::Normal => computed_style(None),
+                LineHeight::Number(value) => (font_size * value.0).to_css_string(),
+                LineHeight::Length(value) => value.0.to_css_string(),
+                LineHeight::MozBlockHeight => content_rect.size.height.to_css_string(),
```

Comment:
Is this `content_rect` the same as the calculated line height used in inline layout?

Reply:
Ah sorry, I suppose I will need to consider it's containing block if it was an inline block too. I added a implementation, but i this #36745 could help with that utilities. I guess I could apply them depends on which one getting merged first.